### DISCUSSION TITLE
test-unicode performance

### DISCUSSION THREAD

**Discussion by crash1013** [nlohmann/json#2682] [General]
I have two Raspberry PI 4's. One with 2 GB RAM and the other 8 GB. I installed the 32-bit Ubuntu server 20.10 on the one with 2GB and the 64-Bit on the one with 8 GB. The one running aarch64 is faster than the one running arm7l in every test except test-unicode. The test doesn't perform well on either system. It runs in half the time on the 32-bit system. Any thoughts on whats going on here?

aarch64 
        Start  96: test-unicode
 96/111 Test  #96: test-unicode .................................   Passed  938.13 sec
        Start  97: test-unicode_valgrind
 97/111 Test  #97: test-unicode_valgrind ........................   Passed    2.22 sec

arm7l
        Start  96: test-unicode
 96/111 Test  #96: test-unicode .................................   Passed  445.07 sec
        Start  97: test-unicode_valgrind
 97/111 Test  #97: test-unicode_valgrind ........................   Passed    2.51 sec
        Start  98: test-user_defined_input

**Comment by nlohmann** [comment]
That test executes a very large test suite (108,551,455 assertions) that includes checking every valid and invalid byte sequence for Unicode characters (8,860,608 tests). I can't explain the differences between these systems, though.

**Comment by crash1013** [comment]
Thanks for your response. I use this library for all JSON related tasks. It is very efficient and robust where I have used it. Everything from intensive logging to application profiles. It is an excellent implementation.
